src/Service/Indexer/SiteKit/QuoteSectionMatcher.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\Search\Service\Indexer\SiteKit;
+
+/**
+ *  @phpstan-type Model array{quote?:string, citation?:string}
+ */
+class QuoteSectionMatcher implements ContentMatcher
+{
+    /**
+     * @inheritDoc
+     */
+    public function match(array $path, array $value): bool|string
+    {
+        $len = count($path);
+        if ($len < 1) {
+            return false;
+        }
+
+        if (
+            $path[$len - 1] !== 'items'
+        ) {
+            return false;
+        }
+
+        if (($value['type'] ?? '') !== 'quote') {
+            return false;
+        }
+
+        $model = $value['model'] ?? false;
+        if (!is_array($model)) {
+            return false;
+        }
+
+        /** @var Model $model */
+
+        $content = [];
+        $quote = $model['quote'] ?? '';
+        if (is_string($quote)) {
+            $content[] = $quote;
+        }
+        $citation = $model['citation'] ?? '';
+        if (is_string($citation)) {
+            $content[] = $citation;
+        }
```

Comment:
`$quote` and `$citation` will always be strings here - according to the `Model` phpstan-type at the top of the class. Either we remove the `is_string` checks or account for those keys to be present while their values are `null`:
```php
/**
 * @phpstan-type Model array{quote?: ?string, citation?: ?string}
 */
```

Reply:
now we can even directly assign the values:

```suggestion
        $content[] = $model['quote'] ?? '';
        $content[] = $model['citation'] ?? '';
```